### PR TITLE
Add validation for name/version with purl in query

### PR DESCRIPTION
**Pull request by SonuBardai**
Closes https://github.com/google/osv.dev/issues/1451

I've added some validations for when the query contains both a `package.purl` and a `package.name`, and when the `package.purl` contains a version and there is a redundant `version` included in the query. In both cases, we're raising a 400 with an error message. Also I've moved the `ecosystem` validation into `do_query` to keep all the validations in one place.

### CODE REVIEW

**Comment by andrewpollock** [comment]
Hey @SonuBardai thank you for your contribution! Once you've got the CLA squared away, I'll take a closer look at this PR.

**Comment by SonuBardai** [comment]
Hey, thanks @andrewpollock. Yeah I've signed the CLA https://github.com/google/osv.dev/pull/1652/checks?check_run_id=17125702423

File: gcp/api/server.py [hunk 1]
```diff
@@ -455,13 +455,29 @@ def do_query(query, context: QueryContext, include_details=True):
...
 
+  if purl and package_name:  # Purls already include the package name
+    context.service_context.abort(
+        grpc.StatusCode.INVALID_ARGUMENT,
+        'name specified in a purl query',
+    )
+  if purl and ecosystem:  # Purls already include the ecosystem inside
+    context.service_context.abort(
+        grpc.StatusCode.INVALID_ARGUMENT,
+        'Ecosystem specified in a purl query',
```
Comment (andrewpollock) [inline gcp/api/server.py:472]: Nit: lowercase Ecosystem for consistency with the other error strings

**Review by andrewpollock — APPROVED** [review]

**Comment by andrewpollock** [comment]
@SonuBardai would you be up for trying to add a test (to https://github.com/google/osv.dev/blob/master/gcp/api/integration_tests.py) for this as well? 

I don't *think* you'll be able to run them locally (but that would be a good test (no pun intended) for an external contributor).

**Comment by SonuBardai** [comment]
@andrewpollock Sure thing! I'll check it out

**Comment by andrewpollock** [comment]
Nicely done with adding the tests!

It looks like:

1) You need to address some findings from our lint and formatting checks, please see https://github.com/google/osv.dev/blob/master/CONTRIBUTING.md#linting-and-formatting
2) Your integration tests aren't working as expected, see https://gist.github.com/andrewpollock/84dd69665fb96fdd436e4128413b6181 for all the gory details

**Comment by SonuBardai** [comment]
Thanks @andrewpollock The traceback from the integration test run was helpful. I was experimenting with how we're checking for the ecosystem in the purl, I think that may have messed with the logic. This new change should fix it.

**Review by andrewpollock — APPROVED** [review]
Thank you for your contribution! 👏 May you make many more!

**Pull request merged**